File: src/main/java/de/storchp/opentracks/osmplugin/download/DownloadMapItem.java
package de.storchp.opentracks.osmplugin.download;

import android.net.Uri;

import androidx.annotation.NonNull;

public record DownloadMapItem(DownloadItemType downloadItemType, String name, String date,
                              String size, Uri uri) implements Comparable<DownloadMapItem> {

    @NonNull
    @Override
    public String toString() {
        return name();
    }

    @Override
    public int compareTo(DownloadMapItem o) {
        if (downloadItemType != o.downloadItemType) {
            return downloadItemType.compareTo(o.downloadItemType);
        }
        return name.compareTo(o.name);
    }
}

File: src/main/java/de/storchp/opentracks/osmplugin/download/DownloadMapItem.kt
package de.storchp.opentracks.osmplugin.download

import android.net.Uri

data class DownloadMapItem(
    val downloadItemType: DownloadItemType,
    val name: String,
    val date: String?,
    val size: String,
    val uri: Uri,
) : Comparable<DownloadMapItem> {
    override fun toString() = this.name

    override fun compareTo(o: DownloadMapItem) =
        if (downloadItemType != o.downloadItemType) {
            downloadItemType.compareTo(o.downloadItemType)
        } else {
            name.compareTo(o.name)
        }

}
